core/src/toga/app.py

```diff
@@ -554,14 +554,6 @@ def _startup(self):
         # This is a wrapper around the user's startup method that performs any
         # post-setup validation.
         self.startup()
-        self._verify_startup()
-
-    def _verify_startup(self):
-        if not isinstance(self.main_window, MainWindow):
-            raise ValueError(
-                "Application does not have a main window. "
-                "Does your startup() method assign a value to self.main_window?"
-            )
```

Comment:
I agree we can't randomly remove it, especially when the behavior on other platforms is not tested. So, should I keep the `_verify_startup()` method and not call it in `_startup()`? Because, this PR is based on the objective that an app can run without a defining a `main_window`.

Reply:
Is that the point of this PR? I thought the point was to decouple the winforms event loop from the App Context. Because that's what the ticket title and description indicate. If that's not the purpose of this PR, then perhaps you should update the ticket title and description to reflect what it is you're trying to do.

Yes, an app without a MainWindow is the obvious way to manifest this problem. And relaxing or changing the contract of toga.App *might* be the outcome of the discussion that is happening on #2108. But that discussion hasn't concluded. It's not yet clear whether there will be a change to the design contract for toga.App, or a different app base class that doesn't have a main window requirement. Proceeding with an implementation when the design hasn't been finalised is extremely premature.

Reply:
The main point of this PR is still to decouple the winforms event loop from the App Context.

To be clear, I am not proposing for removing the usage of `MainWindow` class. I am only suggesting that we should make the defining of `toga.App.main_window` optional. This would allow for apps without any windows(like "StatusIcon-only"apps), without the need for additional app classes.

Reply:
In understand entirely *why* you've done this. What I'm telling you is that it is a major API change that presupposes the outcome of a design discussion that hasn't completed yet. 

You're altering the basic contract of toga.App. As an app author, I can no longer assume that `app.main_window` exists - I need to guard every usage with an `if None` check.

Reply:
I agree with you. There isn't any point on trying to implement something for which the design decision has not been finalized. This PR should be limited to its main objective. If in the future, there would be a requirement for making `main_window` optional then we can create a new PR for accomplishing the same.